### PR TITLE
fix some wrong logic about keep_fnames

### PR DESCRIPTION
**Pull request by gdh1995**
This fix https://github.com/terser/terser/issues/191#issuecomment-540412889 .

### CODE REVIEW

File: lib/scope.js [hunk 1]
```diff
@@ -671,6 +671,7 @@ AST_Toplevel.DEFMETHOD("mangle_names", function(options) {
             && node instanceof AST_VarDef
             && node.value instanceof AST_Lambda
             && !node.value.name
+            && keep_name(options.keep_fnames, node.name.definition().name)
```
Comment (fabiosantoscode) [inline lib/scope.js]: `node.name.name` is a bit faster and shorter too!

**Comment by fabiosantoscode** [comment]
Thanks! I didn't think about regexes when adding the keep_fnames logic to variables.

**Comment by fabiosantoscode** [comment]
I just suggested a change to make the code shorter.

File: lib/scope.js (hunk 1 above)
Comment (gdh1995) [reply lib/scope.js]: updated

**Comment by fabiosantoscode** [comment]
Cool, thanks!

**Pull request merged**